Vue component

marioSantos97 opened the pull request.

marioSantos97 committed:
Merge branch 'master' into vue-component

marioSantos97 committed:
Vue Component Start

Created first templates
Started working on the javaScript integration

marioSantos97 committed:
Selection Feature Completed

Use of jQueryUI for selection feature.
Allow developer to define the look of selected medias UI.
Implemented min and max selected medias feature and feedback
Created base for selected medias edition
FileManager component throws a "save" event whenever a media is selected

marioSantos97 committed:
Added missing features

Asign / Remove tag feature
Refresh Feature
Upload Feature

marioSantos97 committed:
Finished All Features

Finish upload feature;
Added parent select feature;
Added selected media editon feature;
Other fixes

marioSantos97 committed:
Merge branch 'package' into vue-component

marioSantos97 committed:
Fixed linting

Minor fixes

marioSantos97 committed:
Finish Edit Selected Media

Added on validation error feedback

marioSantos97 committed:
Add tags infinite scroll feature

Load paginated tags with infinite scroll ui, both on laravel and vue fields
Replaced deprecated code
Ohter minor fixes

marioSantos97 committed:
Merge branch 'master' into vue-component

marioSantos97 committed:
Add support without media cloud

Save medias in specified disk when no media cloud is in use
Setup parents_filter from config file
Other improves

marioSantos97 committed:
File-Manager works without media cloud

Linting fixed
Other fixes

marioSantos97 committed:
Use JqueryUi for selection feature

marioSantos97 committed:
Media Cloud Integration

File-Manager send a request to media cloud for each file uploaded.

From that request, file-manager gets the file preview  if there is one and create the base record of the media in the database.

File-manager has an endpoint that recieves the paths of generated files and save them in the according media entry.

If there is no preview returned, file-manager will asign a default one.

Some code was improved in the vue component javaScript

promatik commented on src/app/Http/Controllers/API/MediaAPIController.php:
```diff
@@ -190,25 +190,76 @@ public function editMedia(Request $request)
+            ],
+        ]);
+
+        Storage::disk('local')->put(
+            'tmp/' . $request->media->getClientOriginalName(),
+            file_get_contents($request->file('media'))
+        );
```
Não consegues fazer só move ao ficheiro?

marioSantos97 replied:
O problema é conseguir enviá-lo depois para o media cloud no request

promatik commented on src/app/Http/Controllers/API/MediaAPIController.php:
```diff
@@ -190,25 +190,76 @@ public function editMedia(Request $request)
+            file_get_contents($request->file('media'))
+        );
+
+        $response = $client->post(env('MEDIA_CLOUD_ENDPOINT'), [
```
Todas estas variáveis `env`, devem ser configs.

promatik commented on src/resources/js/utils.js:
```diff
@@ -84,9 +84,15 @@ const customEvent = (event, detail = {}, parent = window) => {
   parent.dispatchEvent(e);
 };
 
+const arrayUniqueByKey = (array, key) => [
+  ...new Map(array.map(item =>[item[key], item])).values(),
+]
+
+
```
O ESLint não correu aqui ...

promatik commented on src/public/css/bundle.css:
```diff
@@ -109,6 +109,12 @@
   justify-content: center;
   margin-bottom: 1rem;
 }
+.custom-file-manager .card-body .row .col-sm-3.selectable img {
+  min-width: 4rem;
+  min-height: 4rem;
+  background-image: url("data:image/svg+xml;base64,PHN2ZyB2ZXJzaW9uPSIxLjEiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgd2lkdGg9IjMyIiBoZWlnaHQ9IjMyIiB2aWV3Qm94PSIwIDAgMzIgMzIiPgo8cGF0aCBkPSJNMjkuOTk2IDRjMC4wMDEgMC4wMDEgMC4wMDMgMC4wMDIgMC4wMDQgMC4wMDR2MjMuOTkzYy0wLjAwMSAwLjAwMS0wLjAwMiAwLjAwMy0wLjAwNCAwLjAwNGgtMjcuOTkzYy0wLjAwMS0wLjAwMS0wLjAwMy0wLjAwMi0wLjAwNC0wLjAwNHYtMjMuOTkzYzAuMDAxLTAuMDAx ...
+  background-size: cover;
+}
```
Este bundle.css não está minificado

marioSantos97 committed:
Api fixes

Work without media manager

marioSantos97 committed:
Allow vue and backpack field to work with the same entity

Start to change tags fields to select2 ajax fields

marioSantos97 committed:
Fixed vue component multiple event listeners

marioSantos97 committed:
Tags fields are now Select2 Ajax

Make select2 ajax fields both for vue component and backpack field.

Fixed conflicts when using both vue component and backpack field

marioSantos97 committed:
Multiple Fixes

Fixed local upload when there are no parents

FIxed issue with create/update mediable entities

Upload button showing again when the upload metadata is invalid

Removed unused code

marioSantos97 committed:
Major fixes / New Features

Fixed bug on uploading without parent
Fixed bug with media selection after medias lazy load
Fixed bug with tags filter after tags lazy load
Fixed bug with translatable select2 options
Fixed bug with vue component selection
Automated pre-fill for title and media type when uploading feature

marioSantos97 committed:
Added hability to filter by id to the medias fetch operation
Fetch previously created medias for vue component edition feature.

marioSantos97 committed:
Add fetch operation for parents
Filter parents when user is not admin trait

marioSantos97 committed:
Use "disk" config value to set the location of temporary files

When files are uploaded using media cloud a temporary file is created, now its possible to define the location where this files are stored by changing the config property "disk".
Removed getParents() method, replaced by fetchParents() on MediaCrudController.php.

marioSantos97 committed:
Update MediaAPIController.php

Commit:
Bug Fixed with unset Event Dispatch

It worked fine with when you are creating a single entries, however if multiple records created this event doesnt get fired.

promatik committed:
Merge pull request #7 from vishaltarkar/vue-component

Bug Fixed with unset Event Dispatch

marioSantos97 committed:
Change .env variables to configs
Minify css files on webpack build
